gasprice/lastnbatches.go

```diff
@@ -2,49 +2,52 @@ package gasprice
 
 import (
 	"context"
-	"fmt"
 	"math/big"
 	"sort"
 	"sync"
+
+	"github.com/0xPolygonHermez/zkevm-node/log"
 )
 
 const sampleNumber = 3 // Number of transactions sampled in a batch.
 
-// LastNL2Blocks struct for gas price estimator last n l2 blocks.
-type LastNL2Blocks struct {
+// LastNL2BlocksGasPrice struct for gas price estimator last n l2 blocks.
+type LastNL2BlocksGasPrice struct {
 	lastL2BlockNumber uint64
 	lastPrice         *big.Int
 
 	cfg Config
+	ctx context.Context
 
 	cacheLock sync.RWMutex
 	fetchLock sync.Mutex
 
 	state stateInterface
+	pool  pool
 }
 
-// UpdateGasPriceAvg for last n bathes strategy is not needed to implement this function.
-func (g *LastNL2Blocks) UpdateGasPriceAvg(newValue *big.Int) {}
-
-// NewEstimatorLastNL2Blocks init gas price estimator for last n l2 blocks strategy.
-func NewEstimatorLastNL2Blocks(cfg Config, state stateInterface) *LastNL2Blocks {
-	return &LastNL2Blocks{
+// newLastNL2BlocksGasPriceSuggestor init gas price suggestor for last n l2 blocks strategy.
+func newLastNL2BlocksGasPriceSuggestor(ctx context.Context, cfg Config, state stateInterface, pool pool) *LastNL2BlocksGasPrice {
+	return &LastNL2BlocksGasPrice{
 		cfg:   cfg,
+		ctx:   ctx,
 		state: state,
+		pool:  pool,
 	}
 }
 
-// GetAvgGasPrice calculate avg gas price from last n l2 blocks.
-func (g *LastNL2Blocks) GetAvgGasPrice(ctx context.Context) (*big.Int, error) {
-	l2BlockNumber, err := g.state.GetLastL2BlockNumber(ctx, nil)
+// UpdateGasPriceAvg for last n bathes strategy is not needed to implement this function.
+func (g *LastNL2BlocksGasPrice) UpdateGasPriceAvg() {
+	l2BlockNumber, err := g.state.GetLastL2BlockNumber(g.ctx, nil)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get last l2 block number, err: %v", err)
+		log.Errorf("failed to get last l2 block number, err: %v", err)
 	}
 	g.cacheLock.RLock()
 	lastL2BlockNumber, lastPrice := g.lastL2BlockNumber, g.lastPrice
 	g.cacheLock.RUnlock()
 	if l2BlockNumber == lastL2BlockNumber {
-		return lastPrice, nil
+		log.Debug("Nothing to update")
```

Comment:
please, add more context to the log.

"Block is still the same, no need to update the gas price at the moment"